doc/rados/configuration/bluestore-config-ref.rst

```diff
@@ -369,167 +373,173 @@ throttling
-that expect a significant fraction of relatively small objects.
-
-the 4kb default value aligns well with conventional hdd and ssd devices.  some
-new coarse-iu (indirection unit) qlc ssds however perform and wear best
-when :confval:`bluestore_min_alloc_size_ssd`
-is set at osd creation to match the device's iu:. 8kb, 16kb, or even 64kb.
-these novel storage drives allow one to achieve read performance competitive
-with conventional tlc ssds and write performance faster than hdds, with
-high density and lower cost than tlc ssds.
-
-note that when creating osds on these devices, one must carefully apply the
-non-default value only to appropriate devices, and not to conventional ssd and
-hdd devices.  this may be done through careful ordering of osd creation, custom
-osd device classes, and especially by the use of central configuration _masks_.
-
-quincy and later releases add
-the :confval:`bluestore_use_optimal_io_size_for_min_alloc_size`
-option that enables automatic discovery of the appropriate value as each osd is
-created.  note that the use of ``bcache``, ``opencas``, ``dmcrypt``,
-``ata over ethernet``, `iscsi`, or other device layering / abstraction
-technologies may confound the determination of appropriate values. osds
-deployed on top of vmware storage have been reported to also
-sometimes report a ``rotational`` attribute that does not match the underlying
-hardware.
-
-we suggest inspecting such osds at startup via logs and admin sockets to ensure that
-behavior is appropriate.  note that this also may not work as desired with
-older kernels.  you can check for this by examining the presence and value
-of ``/sys/block/<drive>/queue/optimal_io_size``.
-
-you may also inspect a given osd:
+If these settings are not entered, then the current implementation will
+populate the SPDK map files with kernel file system symbols and will use the
+kernel driver to issue DB/WAL I/Os.
+
+Minimum Allocation Size
+=======================
+
+There is a configured minimum amount of storage that BlueStore allocates on an
+OSD. In practice, this is the least amount of capacity that a RADOS object can
+consume. The configuration option in
+question--:confval:`bluestore_min_alloc_size`--derives its value from the value
+of either :confval:`bluestore_min_alloc_size_hdd` or
+:confval:`bluestore_min_alloc_size_ssd`, depending on the OSD's ``rotational``
+attribute. Thus if an OSD is created on an HDD, BlueStore is initialized with
+the current value of :confval:`bluestore_min_alloc_size_hdd`; but with SSD OSDs
+(including NVMe devices), Bluestore is initialized with the current value of
+:confval:`bluestore_min_alloc_size_ssd`.
+
+In Mimic and earlier releases, the default values were 64KB for rotational
+media (HDD) and 16KB for non-rotational media (SSD). The Octopus release
+changed the the default value for non-rotational media (SSD) to 4KB, and the
+Pacific release changed the default value for rotational media (HDD) to 4KB.
```

Comment:
If you're bored sometime, check out the commit history, it's even more complicated than this.